src/shipit_code_coverage/shipit_code_coverage/utils.py

```diff
@@ -49,6 +49,11 @@ def submit(self, *args, **kwargs):
         return future
 
     def __exit__(self, *args):
-        for future in concurrent.futures.as_completed(self.futures):
-            future.result()
+        try:
+            for future in concurrent.futures.as_completed(self.futures):
+                future.result()
+        except Exception as e:
+            for future in self.futures:
+                future.cancel()
```

Comment:
I imagine that this might also `.cancel()` `future`s that have already succeeded, but maybe that's not a problem.

Reply:
Yeah it should basically be a no-op in that case.